pkg/kubelet/dockertools/docker.go

```diff
@@ -160,6 +160,10 @@ func (p dockerPuller) Pull(image string, secrets []api.Secret) error {
 
 		err := p.client.PullImage(opts, docker.AuthConfiguration{})
 		if err == nil {
+			// Sometimes PullImage failed with no error returned.
```

Comment:
Please differentiate between the `!exist` and `ierr != nil` cases, or just print the `ierr` into the error message